### PR TITLE
#167190438 User's Profile

### PR DESCRIPTION
**Pull request by Hadeneekeh**
####  What does this PR do?
- It has `GET /api/user` which creates the user's profile
- It has `PUT /api/user` which edits the user's profile
- It has `GET /api/profiles/:username` which has viewing another user's profile

#### Description of Task to be completed?
- write test
- Created the following endpoints:
     * get authorised user's profile
     * edit authorised user's profile
     * view any user's profile
- document the feature on swagger

#### How should this be manually tested?
- fetch or pull `ft/167190438/create-user-profile`
- run npm install to get the packages used
- Test the endpoint above using postman as demonstrated in the screenshots below

#### Any background context you want to provide?
- use the string `'image'` as the key for the image upload.

#### What are the relevant pivotal tracker stories?
[#167190438](https://www.pivotaltracker.com/story/show/167190438)

#### Screenshots (if appropriate)
image

#### Questions:
NA

### CODE REVIEW

File: server/controllers/userController.js [hunk 1]
```diff
@@ -83,6 +85,78 @@ class UserController {
...
+  * @static
+  * @description Edit user's profile
+  * @param {Object} req - Request object
+  * @param {Object} res - Response object
+  * @returns {Object} object containing the user's profile
+  * @memberof UserController
+  */
+  static async editProfile(req, res) {
+    const { id } = req.loggedInUser;
+    const { bio, email, username } = req.body;
```
Comment (chuxmykel) [inline server/controllers/userController.js]: Great Job @Hadeneekeh. I actually knew you could pull this off. I'd love to have a sync with you later to understand how you implemented the image upload successfully. I have a question though, how about if as a user, I want to update just my `email` or `username`? Can this handle it?

Comment (Hadeneekeh) [reply server/controllers/userController.js]: Thank you for asking. Yes, it will update the database with only the input that you supplied.

**Review by cvjude — COMMENTED** [review]
LGTM

**Review by encodedBicoding — APPROVED** [review]
LGTM

**Review by chiboycalix — APPROVED** [review]
LGTM

**Review by topseySuave — APPROVED** [review]

**Comment by topseySuave** [comment]
Please next time try to reduce the amount similar commits you make

**Pull request merged**